src/combinations.rs

```diff
@@ -53,7 +101,7 @@ impl<I> Iterator for Combinations<I>
     type Item = Vec<I::Item>;
     fn next(&mut self) -> Option<Self::Item> {
         if self.first {
-            if self.pool.is_done() {
+            if self.pool.is_done() && self.k() > self.n() {
```

Comment:
Could you elaborate why this became necessary?

Reply:
It seems that previously the combined condition `first && pool.done` served to indicate that `k` was larger than `n` and that no items would be yielded. With my changes this is no longer the case. Now when `Combinations` is reused internally within `Powerset` `pool.done` is always `true`, even during the first call to `next()`, since we are working with the same pool data.

There are other ways to do this, but this way seemed to make the most sense.

Reply:
Couldn't we then omit the `self.pool.is_done` check?

Reply:
Yes, now removed.